File: security-test-suite/java/vulnerable/sql_injection.java
import java.io.*;
import java.sql.*;

public class VulnerableSQL {
    public void queryDatabase(String username) throws Exception {
        Connection conn = DriverManager.getConnection("jdbc:mysql://localhost:3306/db", "root", "password");
        Statement stmt = conn.createStatement();
        
        // VULNERABLE: String concatenation
        String query = "SELECT * FROM users WHERE username = '" + username + "'";
        stmt.executeQuery(query);
    }
}

File: security-test-suite/python/vulnerable/sql_injection.py
import sqlite3

def query_database(username):
    conn = sqlite3.connect('example.db')
    cursor = conn.cursor()
    
    # VULNERABLE: String formatting
    query = "SELECT * FROM users WHERE username = '%s'" % username
    cursor.execute(query)
    
    # VULNERABLE: f-string
    cursor.execute(f"SELECT * FROM users WHERE username = '{username}'")
